File: 2017 - 2/Programação de Computadores/All Netbeans Projects/CPPJogo/Tempo.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Tempo.cpp
 * Author: 743554
 * 
 * Created on 31 de Agosto de 2017, 13:38
 */


#include <iostream>
#include <cstdio>
#include <stdlib.h>


using namespace std;


Tempo Tempo::diferenca(Tempo* outro) {
    
    int difM = outro->minuto - this->minuto;
    int difH = outro->hora - this->hora;    
    
    if(difM > 0){       
        
        
    }

    
}

void Tempo::imprime(){
    
    cout << hora "Hora(s) e " << minuto <<" minuto(s)" << endl;
    printf("%d Hora(s) e %d minuto(s)\n", hora, minuto);
}

File: 2017 - 2/Programação de Computadores/Delano fez/Java/Jogo/src/Tempo.java

public class Tempo {

    private int hora;
    private int minuto;

    public Tempo(int hora, int minuto) {
        this.hora = hora;
        this.minuto = minuto;
    }

    Tempo diferenca(Tempo outro) {
        int difM = outro.minuto - this.minuto;
        int difH = outro.hora - this.hora;
        if (difM < 0) {
            difM += 60;
            difH -= 1;
        }

        if (difH < 0 || (difH == 0 && difM == 0)) {
            difH += 24;
        }

        return new Tempo(difH, difM);
    }

    void imprime() {
        System.out.println(hora + " hora(s) e " + minuto + " minuto(s)");
    }
}
